eth/fetcher/block_fetcher.go

```diff
@@ -330,11 +360,15 @@ func (f *BlockFetcher) loop() {
 				break
 			}
 			// Otherwise if fresh and still unknown, try and import
-			if number+maxUncleDist < height || f.getBlock(hash) != nil {
+			if number+maxUncleDist < height || f.light && f.getHeader(hash) != nil || !f.light && f.getBlock(hash) != nil {
```

Comment:
Please add parentheses around the `x && y` groups. I know Go's convention is that `&&` has a higher precedence than `||`, but I really don't want to do mental gymnastics to figure out what comes before what else :)